pkgs/games/minetest/mods/build-minetest-mod.nix

```diff
@@ -0,0 +1,33 @@
+{stdenv}:
+
+rec {
+  buildMinetestMod =
+    attrs@{
+      name ? "${attrs.pname}",
+      namePrefix ? "",
+      src,
+      unpackPhase ? "",
+      configurePhase ? "",
+      buildPhase ? "",
+      preInstall ? "",
+      postInstall ? "",
+      path ? ".",
+      addonInfo ? null,
+      ...
+    }:
+    stdenv.mkDerivation (attrs // {
+        name = namePrefix + name;
+        nativeBuildInputs = [];
+        inherit unpackPhase configurePhase buildPhase addonInfo preInstall postInstall;
```

Comment:
Why preInstall and postInstall? Isn't `attrs //` enough?

Reply:
That part was cargo-culted quite a while ago from the builder for vim plugin derivations. https://github.com/nix-community/home-manager/blob/master/modules/programs/vim.nix 

ATM it seems to me that the whole `inherit` line should be dispensable since it will only cause identical overriding.

Reply:
I think `*Phase` inherits are needed if we want to skip the default ones (and still allow pre/post, so no `dont*`)

Reply:
But the inherits apparently are redundant, that's my point. `inherit preInstall` means `preInstall = preInstall`. `preInstall`on the right of the assignment is `attrs.preInstall`. And then something like `attrs // {preInstall = attrs.preInstall;}` which is just `attrs`.

Am I missing anything? As I said, I copy/pasted this part from https://github.com/nix-community/home-manager/blob/master/modules/programs/vim.nix, which does a similar job for vim plugins, and didn't bother much anymore once it worked.